sigopt/xgboost/experiment.py

```diff
@@ -66,19 +77,51 @@ def parse_and_create_metrics(self):
         else:
           metric['name'] = DEFAULT_EVALS_NAME + '-' + metric['name']
 
+  def check_and_fill_parameter_types(self):
+    params_to_check = [p for p in self.experiment_config_parsed['parameters'] if p['name'] in PARAMETER_INFORMATION]
+    for parameter in params_to_check:
+      parameter_name = parameter['name']
+      if parameter_name in PARAMETER_INFORMATION:  # TODO: if parameter_name isn't here, probably should throw warning
```

Comment:
this `if` statement is redundant after line 81. You should remove one.

Reply:
I suggest leave this one here and remove the one in 81. You can add an else statement here
```
else:
  if 'type' note in parameters:
    raise ValueError(.....)
```

Reply:
I prefer less control flow if necessary so I removed 84